gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -2568,8 +2563,8 @@ public static String GET_SNAPSHOT_TXID() throws SQLException, StandardException
     }
     // tc commit will clear all the artifacts but will not commit actual txState
     // that should be committed in COMMIT procedure
-    tc.resetActiveTXState(true);
-    TXManagerImpl.getOrCreateTXContext().clearTXState();
+    tc.clearActiveTXState(true, true);
+    //tc.resetActiveTXState(true);
```

Comment:
@suranjan Can think of two ways. One is a ThreadLocal. Other cleaner way can be that when doing a zipPartitions in ColumnFormatRelation, the iterator handles are available so it can cast the left row buffer iterator and get the current connection handle, then set into the column iterator. The conn on latter will become a var that will be set from outside rather than using helper and the preparedStatement a lazy val. Likewise the context listener will use iterator.conn rather than the one from helper.